flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/filemapping/FileMappingManager.java

```diff
@@ -231,14 +270,23 @@ private boolean isParentDir(String path, String dir) {
         }
     }
 
-    /** A wrapper of real path. */
-    public static class RealPath {
-        public final Path path;
-        public final boolean isLocal;
+    public void giveUpOwnership(Path path) {
+        MappingEntry mappingEntry = mappingTable.getOrDefault(path.toString(), null);
+        Preconditions.checkArgument(
+                mappingEntry != null,
+                "Try to give up ownership of a file that is not in mapping table: " + path);
+        Preconditions.checkArgument(
+                mappingEntry.fileOwnership != FileOwnership.PRIVATE_OWNED_BY_DB,
+                "Try to give up ownership of a file that is not shareable: " + mappingEntry);
```

Comment:
Please use the string template version of `checkArgument` here, to avoid much string generation overhead.